### PR TITLE
[google_maps_flutter] Updates platform interface to new analysis options

### PR DESCRIPTION
**Pull request by stuartmorgan**
Converts to the new analysis options, fixing all violations.

Since this was the last package on the legacy options, also removes the legacy options file, as well as references to it, and updates the repo tooling to allow a config file that is empty. (The config file itself can't be removed, since there are out-of-repo scripts referencing it.)

Part of https://github.com/flutter/flutter/issues/76229

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by ditman — APPROVED** [review]
Amazing, LGTM!

**Comment by ditman** [comment]
Wouldn't this close https://github.com/flutter/flutter/issues/76229 ?

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart [hunk 1]
```diff
@@ -6,24 +6,70 @@ import 'dart:async' show Future;
...
 /// image to place on the surface of the earth.
 class BitmapDescriptor {
   const BitmapDescriptor._(this._json);
 
+  /// The inverse of .toJson.
+  // This is needed in Web to re-hydrate BitmapDescriptors that have been
+  // transformed to JSON for transport.
+  // TODO(stuartmorgan): Clean this up. See
+  // https://github.com/flutter/flutter/issues/70330
+  BitmapDescriptor.fromJson(Object json) : _json = json {
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart:25]: I think this one may actually be legacy after my first cleanup. After we do the rest of the cleanup we should mark things weren't not using ourselves now as deprecated.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart [hunk 2]
```diff
@@ -40,16 +42,19 @@ class LatLng {
       return null;
     }
     assert(json is List && json.length == 2);
-    final list = json as List;
-    return LatLng(list[0], list[1]);
+    final List<dynamic> list = json as List<dynamic>;
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart:45]: Yes, fixed. I think I just got burned enoughby careless removal of `dynamic` from plugin code early on that I'm still slightly nervous about it :)

**Comment by stuartmorgan** [comment]
> Wouldn't this close [flutter/flutter#76229](https://github.com/flutter/flutter/issues/76229) ?

It would have, but we got so far behind flutter/flutter's list of analysis options during this transition that I'm now tracking catching up there. Then I'll close it.

**Pull request merged**